stan/math/prim/scal/prob/normal_rng.hpp

```diff
@@ -1,39 +1,66 @@
 #ifndef STAN_MATH_PRIM_SCAL_PROB_NORMAL_RNG_HPP
 #define STAN_MATH_PRIM_SCAL_PROB_NORMAL_RNG_HPP
 
-#include <boost/random/normal_distribution.hpp>
-#include <boost/random/variate_generator.hpp>
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
-#include <stan/math/prim/scal/err/check_positive.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/scal/err/check_positive_finite.hpp>
+#include <stan/math/prim/scal/meta/VectorBuilder.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
```

Comment:
The doc is a little unclear at the edges, much like C++.  There are only one or two things I really hated, like the odd-numbered indents in some class declaration stuff.

The big departure is that we allow references and exceptions.  Google doesn't like either one because they make a huge code base harder to understand from the outside (you don't know what exception a function might throw and having pointers helps distinguish what's going to get passed).